backend/dissemination/management/commands/scan_bucket_files_for_viruses.py

```diff
@@ -0,0 +1,145 @@
+# This is file-like, and can be used for passing on
+# to a POST to ClamAV.
+def load_file_from_s3(bucket, object_name):
+    s3 = get_s3_client()
+    file = BytesIO()
+    try:
+        s3.download_fileobj(bucket, object_name, file)
+    except ClientError:
+        logger.error("======================================")
+        logger.error(f"Could not download {object_name}")
+        logger.error("======================================")
+        return None
+    except Exception as e:
+        logger.error("======================================")
+        logger.error(f"{e}")
+        logger.error("======================================")
+        return None
+
+    # Seek to the start of the file.
+    file.seek(0)
+    return file
+
+
+def scan_file_in_s3(bucket, object_name):
+    try:
+        io_obj = load_file_from_s3(bucket, object_name)
+        bytes_blob = io_obj.read()
+        simple_obj = SimpleUploadedFile(object_name, bytes_blob)
+        return _scan_file(simple_obj)
+    except Exception as e:
+        logger.error(f"SCAN SCAN_FILE_IN_S3 {e}")
+        return f"{object_name}"
+
+
+def scan_files_at_path_in_s3(bucket, path):
+    s3 = get_s3_client()
+    objects = s3.list_objects(Bucket=bucket, Prefix=path)
+    if objects:
+        results = []
+        if "Contents" in objects:
+            for object_summary in objects["Contents"]:
+                object_name = object_summary["Key"]
+                result = scan_file_in_s3(bucket, object_name)
+                results.append(result)
+            return results
+    return None
+
+
+def is_stringlike(o):
+    return isinstance(o, str) or isinstance(o, bytes)
+
+
+def not_a_stringlike(o):
+    return not is_stringlike(o)
+
+
+def check_scan_ok(result):
+    if result and not_a_stringlike(result) and result.status_code == 200:
+        return True
+    else:
+        return False
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument("--bucket", type=str, required=True)
+        parser.add_argument("--object", type=str, required=False, default=None)
+        parser.add_argument("--path", type=str, required=False, default=None)
+        pass
+
+    def handle(self, *args, **options):
+        bucket = options["bucket"]
+        object = options["object"]
+        path = options["path"]
+
+        # Do we want to scan a single object?
+        if object:
+            result = scan_file_in_s3(bucket, object)
+            if check_scan_ok(result):
+                pass
+            else:
+                logger.error(f"SCAN FAIL: {object}")
+        if path:
+            results = scan_files_at_path_in_s3(bucket, path)
+            if results:
+                if all(map(check_scan_ok, results)):
+                    logger.info(f"SCAN OK: COUNT {len(results)}")
+                else:
+                    for r in results:
+                        if is_stringlike(r):
+                            logger.error(f"SCAN FAIL: {r}")
```

Comment:
As this is right now, since this is just a django command, and has no associated workflow, (and even if it did, question still applies when run with cf tasks) how do we get the output back to us for consumption? 

Could we write fails to a file and send it off somewhere?

Reply:
I thought that the `logger` actions, as a Django command, would be picked up by `logshipper`/NR. 

Is that incorrect? 

If so, then yes: I need to bundle up results and send them somewhere.

Reply:
According to https://docs.cloudfoundry.org/devguide/using-tasks.html#-task-logging-and-execution-history :

> Any data or messages the task outputs to stdout or stderr is available in the firehose logs of the app. A syslog drain attached to the app receives the task log output. The task execution history is retained for one month.

So it looks like that `logger.info` line would be picked up by `logshipper`/NR.